src/Amazon.AspNetCore.Identity.Cognito/CognitoSigninManager.cs

```diff
@@ -163,5 +208,102 @@ protected Task<bool> IsPasswordChangeRequiredAsync(TUser user)
         {
             return _userManager.IsPasswordChangeRequiredAsync(user);
         }
+
+        /// <summary>
+        /// Validates the two factor sign in code and creates and signs in the user, as an asynchronous operation.
+        /// </summary>
+        /// <param name="code">The two factor authentication code to validate.</param>
+        /// <param name="isPersistent">Flag indicating whether the sign-in cookie should persist after the browser is closed.</param>
+        /// <param name="rememberClient">Flag indicating whether the current browser should be remember, suppressing all further 
+        /// two factor authentication prompts.</param>
+        /// <returns>The task object representing the asynchronous operation containing the <see name="SignInResult"/>
+        /// for the sign-in attempt.</returns>
+        public async Task<SignInResult> RespondToTwoFactorChallengeAsync(string code, bool isPersistent, bool rememberClient)
+        {
+            var twoFactorInfo = await RetrieveTwoFactorInfoAsync().ConfigureAwait(false);
+            if (twoFactorInfo == null || twoFactorInfo.UserId == null)
+            {
+                return SignInResult.Failed;
+            }
+            var user = await _userManager.FindByIdAsync(twoFactorInfo.UserId).ConfigureAwait(false);
+            if (user == null)
+            {
+                return SignInResult.Failed;
+            }
+
+            // Responding to the Cognito challenge.
+            await _userManager.RespondToTwoFactorChallengeAsync(user, code, twoFactorInfo.CognitoAuthenticationWorkflowId).ConfigureAwait(false);
+
+            if (user.SessionTokens != null && user.SessionTokens.IsValid())
```

Comment:
nit: considering inverting the conditional.  In this method, most if blocks check for failure condition and return SignInResult.Failed.  This particular block does the opposite which is a bit weird.